crypto/ct/ct_b64.c

```diff
@@ -45,10 +45,14 @@ static int ct_base64_decode(const char *in, unsigned char **out)
         goto err;
     }
 
-    /* Subtract padding bytes from |outlen| */
-    while (in[--inlen] == '=') {
+    /* Subtract padding bytes from |outlen|.  Any more than 2 is malformed. */
+    i = 0;
+    while (in[--inlen] == '=' && i < 2) {
         --outlen;
+        ++i;
     }
+    if (i > 2)
+        goto err;
```

Comment:
It will never be more than 2, you're breaking the loop before that can happen.  I rather think that the loop itself will have to look something like this:

``` C
    while (in[--inlen] == '=') {
        --outlen;
        if (++i > 2)
            goto err;
    }
```